horovod/_keras/__init__.py

```diff
@@ -73,8 +79,13 @@ def init_aggregation_vars():
                             elif isinstance(grad, tf.IndexedSlices):
                                 raise AssertionError(
                                     "IndexedSlices are not supported when `self._aggregation_frequency` > 1 and `self._sparse_as_dense is False")
+                            if self.grad_updated_sizes_dict:
+                                assert idx in self.grad_updated_sizes_dict
```

Comment:
technically this probably doesn't do what you want, since you access the element in the dict with `str(idx)`, just gotta pick which approach you want. A best practice would probably be something like:

```if idx not in self.grad_updated_sizes_dict:
    raise AssertionError("Gradient size not in ____")
```

I also am not a huge fan of the name :/ but can let that go

Reply:
the things ive done here ...